### DISCUSSION TITLE
Tab styling - escape seq's or similar?

### DISCUSSION THREAD

**Discussion by alaricljs** [kovidgoyal/kitty#5329] [Q&A] [answered]
Beyond glyphs and basic fg/bg styling - I'd like to get underlining in my tab and am failing.

My goal is something as close to this as I can manage:
![1-region](https://user-images.githubusercontent.com/5070475/181598423-8042b981-fdc3-4919-b285-c434914c0223.jpg)

The primary thing I'm working on right now is getting that line in the active tab rendered.

This is the only thing I've attempted, which does not work: 
```python
def draw_tab( ... ) -> int:

    screen.draw(f'\e[4:1mTab Title\e[4:0m')

```

As you might imagine this prints exactly what's there, no escape interpretation and hence no underline.  I haven't discovered anything in the code or docs that I've been able to interpret pointing me in the right direction.

Is what I'm looking to achieve possible?  How?

**Comment by page-down** [comment, marked as answer]
There is a `draw_attributed_string` function in tab_bar.py for reference, or just use it.

https://github.com/kovidgoyal/kitty/blob/v0.25.2/kitty/tab_bar.py#L155-L163
